gwcs/wcs.py

```diff
@@ -634,3 +641,269 @@ def fix_inputs(self, fixed):
         new_pipeline.append((step0[0], new_transform))
         new_pipeline.extend(self.pipeline[1:])
         return self.__class__(new_pipeline)
+
+    def to_fits_sip(self, bounding_box, max_pix_error=0.25, degree=None,
+                    max_inv_pix_error=0.25, inv_degree=None, 
+                    npoints=32, verbose=False):
+        """
+        Construct a SIP-based approximation to the WCS in the form of a FITS header
+
+        This assumes a tangent projection.
+
+        The default mode in using this attempts to achieve roughly 0.25 pixel
+        accuracy over the whole image.
+
+        Parameters
+        ----------
+        bounding_box : a pair of tuples, each consisting of two integers
+            Represents the range of pixel values in both dimensions
+            ((xmin, xmax), (ymin, ymax))
+        max_pix_error : float, optional
+            Maximum allowed error over the domain of the pixel array. This
+            error is the equivalent pixel error that corresponds to the maximum
+            error in the output coordinate resulting from the fit based on
+            a nominal plate scale.
+        degree : int, optional
+            Degree of the SIP polynomial. If supplied, max_pixel_error is ignored.
+        max_inv_error : float, optional
+            Maximum allowed inverse error over the domain of the pixel array
+            in pixel units. If None, no inverse is generated.
+        inv_degree : int, optional
+            Degree of the inverse SIP polynomial. If supplied max_inv_pixel_error
+            is ignored.
+        npoints : int, optional
+            The number of points in each dimension to sample the bounding box
+            for use in the SIP fit.
+        verbose : bool, optional
+            print progress of fits
+
+        Returns
+        -------
+        FITS header with all SIP WCS keywords
+
+        Raises
+        ------
+        ValueError
+            If the WCS is not 2D, an exception will be raised. If the specified accuracy
+            (both forward and inverse, both rms and maximum) is not achieved an exception
+            will be raised.
+
+        Notes
+        -----
+
+        Use of this requires a judicious choice of required accuracies. Attempts to use
+        higher degrees (~7 or higher) will typically fail due floating point problems
+        that arise with high powers.
+
+
+        """
+        if (self.forward_transform.n_inputs != 2
+                or self.forward_transform.n_outputs != 2):
+            raise ValueError(
+                "The to_fits_sip method only works with 2-dimensional transforms")
+
+        transform = self.forward_transform
+        # Determine reference points.
+        (xmin, xmax), (ymin, ymax) = bounding_box
+        crpix1 = int((xmax - xmin) / 2)
+        crpix2 = int((ymax - ymin) / 2)
```

Comment:
There's no reason for these to be integers.

Reply:
No, but it does make it simpler to interpret the array data. Is there any practical purpose of not making them integers from the bounding box? If it came from a crvals, yes, they shouldn't be integers, but here?

Reply:
I put the default values as placeholders to keep the FITS header keywords in that order. But sure I can change that, but these values are always changed.

Reply:
I think here it is fine to have integers. Indeed it is easier to look at the header and see 1020 instead of 1019.95678. (They are still written out as floats though (I think))

Reply:
Agreed.

Reply:
One possible variation (depending on tastes) is: `crpix1 = (xmax - xmin) // 2` (because `crpix` *in the `astropy.wcs.WCS.WcsPrm`* is not an `int`.

Reply:
changed to use integer division